graph.c

```diff
@@ -253,16 +253,11 @@ void graph_free(Graph **self) {
 
 int* graph_init_vertices_indexes() {
     //static arr also can be used but not so explicit
-    int* arr = (int*)malloc(GRAPH_SIZE * sizeof(int));
-
-    if (arr == NULL) {
-        printf("Memory allocation failed!\n");
-        return NULL;
-    }
+    static int vertices_indexes[GRAPH_SIZE];
```

Comment:
Why did you put static here?

Also I'd suggest to make graph size dynamic, you'll write various algorithms on top of graph, so hardcoding it's size is a bag idea